migrations/20230613116515_courses.sql

```diff
@@ -0,0 +1,103 @@
+-- Courses Table
+CREATE TABLE IF NOT EXISTS public.courses (
+    id SERIAL PRIMARY KEY,
+    catalog_number SMALLINT NOT NULL,
+    subject_code VARCHAR(10) NOT NULL,
+    external_id VARCHAR(10) NOT NULL,
+    academic_level VARCHAR(13) NOT NULL,
+    title TEXT NOT NULL,
+    description TEXT NOT NULL,
+    abbreviated_description TEXT NOT NULL,
+    requirements_description TEXT NOT NULL,
+    prerequisites_id INTEGER NULL,
+    CONSTRAINT uk_courses_external_id UNIQUE (external_id),
+    CONSTRAINT fk_courses_to_prerequisites FOREIGN KEY (prerequisites_id)
+    REFERENCES prerequisites (id),
+    CONSTRAINT chk_courses_external_id_not_empty CHECK (external_id <> ''),
+    CONSTRAINT chk_courses_subject_code_not_empty CHECK (subject_code <> ''),
+    CONSTRAINT chk_courses_academic_level CHECK (academic_level IN ('undergraduate', 'graduate')),
+    CONSTRAINT chk_courses_title_not_empty CHECK (title <> '')
+);
+
+-- Courses Table Indices
+CREATE INDEX CONCURRENTLY IF NOT EXISTS idx_courses_subject_code_catalog_number
```

Comment:
Best guess on which column combinations will be most frequently in the selection criteria (common queries).